service/templates/search.html

```diff
@@ -19,12 +19,21 @@ <h1 class="heading-xlarge form-title">Find a title</h1>
                           <input type="text" class="form-control" name="search_term" id="search_term">
                       </div>
                       <div class="form-group">
-                          <input type="submit" value="Search" class="button">
+                          <input id="search_button" type="submit" value="Search" class="button">
                       </div>
                   </form>
                 </div>
             </div>
-
         </div>
     </main>
+    <script type="text/javascript">
+        function ga_title_search() {
+            search_term = document.getElementById('search_term').value
+            title_number_regex = '^([A-Z]{0,3}[1-9][0-9]{0,5}|[0-9]{1,6}[ZT])$'
+            if (search_term.match(title_number_regex)) {
+              ga('send', 'event', 'Search', 'TitleNumber', search_term);
+            }
+        }
+        document.getElementById('search_button').onclick = ga_title_search;
```

Comment:
a) What I proposed is to keep js separately from html. Nothing really changes. 

b) The question was really about whether we know that the element already exists when we're trying to assign a handler to its event. If we try to do it before it gets loaded, there will be no event handling. As far as I remember, when using jQuery, the standard place to do such things would be in response to document.ready. I'm not an expert, so just asking.